Code/Framework/AzCore/AzCore/Memory/AllocatorBase.cpp

```diff
@@ -119,30 +256,29 @@ bool AllocatorBase::IsProfilingActive() const
     return m_isProfilingActive;
 }
 
-void AllocatorBase::DisableOverriding()
-{
-    m_canBeOverridden = false;
-}
-
 void AllocatorBase::DisableRegistration()
 {
     m_registrationEnabled = false;
 }
 
 void AllocatorBase::ProfileAllocation(void* ptr, size_t byteSize, size_t alignment, const char* name, const char* fileName, int lineNum, int suppressStackRecord)
 {
+    if (m_isProfilingActive)
+    {
 #if defined(AZ_HAS_VARIADIC_TEMPLATES) && defined(AZ_DEBUG_BUILD)
```

Comment:
The `AZ_HAS_VARIADIC_TEMPLATES` define doesn't even make any sense here.
Also it is not needed since we have had support since VS2013

Reply:
removing the #if since it is no longer an EBus call